docs(mcp): add MCP how to workflow guides (#1964)
IgniteUI/igniteui-xplat-docs: doc/en/components/ai/cli-mcp.md

---

> title: Ignite UI MCP Overview | {ProductName} | Infragistics

Reviewer: ```suggestion
title: {Platform} Ignite UI MCP Overview - {ProductName}
```

> _description: Learn what Ignite UI MCP is and choose the right {Platform} MCP workflow topic to start with.

Reviewer: ```suggestion
_description: Ignite UI MCP is a set of Model Context Protocol servers that expose Ignite UI CLI and theming tools to AI-assisted editors. Choose the right setup path for {Platform}.
```

> _keywords: {Platform}, {ProductName}, Infragistics, MCP, Model Context Protocol, Ignite UI MCP, AI, Ignite UI CLI MCP, Ignite UI Theming MCP

Reviewer: you can add: agent

> _language: en

Reviewer: ```suggestion
_language: en
_license: MIT
```

mentionedTypes: []
---

> # Ignite UI CLI MCP Overview

Reviewer: ```suggestion
# {Platform} Ignite UI CLI MCP Overview
```

> This topic is the quick entry point for Ignite UI MCP for {Platform}.

Reviewer: ```suggestion
Ignite UI CLI MCP and Ignite UI Theming MCP are two standalone Model Context Protocol servers that expose Ignite UI for {Platform} CLI and theming tools to AI-assisted editors and chat clients — including VS Code with GitHub Copilot, Cursor, Claude Desktop, Claude Code, and JetBrains IDEs with AI Assistant. The two servers (`igniteui-cli` and `igniteui-theming`) run via `npx` and connect to any MCP-compatible client through STDIO transport. Once connected, the AI assistant can create {Platform} projects, add and modify Ignite UI for {Platform} components, apply themes, and answer documentation and API questions — all through natural-language prompts.
```

> Use it to understand what Ignite UI MCP is, how it fits into your editor or AI client, and which setup path to open next.

Reviewer: ```suggestion
Neither server modifies project files autonomously — each server exposes tools to the active AI agent session in a supported client. Ignite UI CLI MCP does not replace Ignite UI CLI commands; it makes them available to the AI client through the MCP protocol. MCP clients that use HTTP-based transport instead of STDIO are not supported by the current server configuration.
```

## What Ignite UI MCP Is

Ignite UI MCP is a set of MCP servers that make Ignite UI tools available to an AI assistant in a supported editor or chat experience.

The main servers used in the current setup flows are:

- `igniteui` for creating projects, modifying and editing existing projects, creating and changing components, and asking documentation questions
- `igniteui-theming` for theme and styling-related work

Together, they help the assistant create a {Platform} project, continue setup work, support theming tasks, and answer product questions from chat.

Supported AI clients can include VS Code with GitHub Copilot, Cursor, Claude Desktop, Claude Code, JetBrains IDEs with AI Assistant, and other MCP-compatible clients.

> The exact setup flow can vary a little depending on your editor and AI client.

Reviewer: ```suggestion
## What Ignite UI CLI MCP and Ignite UI Theming MCP Provide

Ignite UI CLI MCP and Ignite UI Theming MCP each expose a distinct set of tools to the AI client.

Ignite UI CLI MCP handles project and component work: creating new {Platform} projects, adding and modifying Ignite UI for {Platform} components, updating existing project structure, and answering documentation and API questions about {Platform} components, properties, events, and methods.

Ignite UI Theming MCP handles theme and styling work: applying and modifying Ignite UI themes, generating CSS custom property overrides, and answering theming-related questions.

Both servers are started through `npx` using commands similar to these:

- `igniteui-cli`: `npx -y igniteui-cli@next mcp`
- `igniteui-theming`: `npx -y igniteui-theming igniteui-theming-mcp`

The `-y` flag tells `npx` to confirm the package download automatically so the servers start without manual intervention.```

## Before You Start

Before you configure Ignite UI MCP, make sure you have:

- Node.js installed so `npx` is available
- a supported editor or AI client with MCP support
- internet access for package resolution when the MCP servers start

Some workflows also assume you already have a {Platform} project or that you want to create one in a new empty folder.

> 

Reviewer: Move the paragraph as #2 right after the Overview sction

## How the MCP Setup Works

The Ignite UI MCP servers are started through `npx`.

In practice, the setup is always the same at a high level:

- add MCP configuration for your AI client
- point that configuration to the Ignite UI MCP servers
- let the client load the servers and expose their tools
- continue in chat or agent mode

The server entries use a command and arguments similar to these:

- `igniteui`: `npx -y igniteui-cli@next mcp`
- `igniteui-theming`: `npx -y igniteui-theming igniteui-theming-mcp`

The `-y` flag tells `npx` to auto-confirm the package download prompt so the servers can start without manual intervention.

> 

Reviewer: ```suggestion
## How the MCP Setup Works

The setup process is consistent across all supported clients and all three setup paths.

Add the Ignite UI CLI MCP and Ignite UI Theming MCP server entries to the configuration file for your AI client. The client reads that configuration, starts the servers through `npx`, and exposes their tools in chat or agent mode. From that point, prompts in the chat session can invoke the tools from either server directly.

The configuration file location and JSON structure differ by client, but the server command and arguments are the same across all clients. The step-by-step topics below show the exact file locations and JSON structure for each client.
```

> ## Editors and AI Clients

Reviewer: ```suggestion
## Supported AI Clients
```

Ignite UI MCP is not limited to one editor or one chat provider.

You can use it with different combinations of editors and AI clients, as long as they support MCP.

> Common examples include:

Reviewer: ```suggestion
Ignite UI CLI MCP and Ignite UI Theming MCP work with any editor or AI client that supports MCP with STDIO transport. Supported clients include:
```

- **VS Code with GitHub Copilot** using a workspace-level `.vscode/mcp.json`
- **Cursor** using `.cursor/mcp.json`

> - **Claude Desktop** using the desktop client MCP configuration file

Reviewer: ```suggestion
- **Claude Desktop** — macOS: `~/Library/Application Support/Claude/claude_desktop_config.json`  
- **Claude Desktop** — Windows: `%APPDATA%\Claude\claude_desktop_config.json`
```

- **Claude Code** using `.mcp.json` or the `claude mcp` command

> - **JetBrains IDEs with AI Assistant** using the IDE MCP settings

Reviewer: ```suggestion
- **JetBrains IDEs with AI Assistant** — configured in **Tools → AI Assistant → Model Context Protocol (MCP)**
```

- **Other MCP clients** using the same command and arguments through STDIO configuration

In practice, the main differences are usually:

- where the MCP configuration is stored
- how the client shows the available MCP tools
- whether you continue in chat, agent mode, or another AI panel

The Ignite UI MCP servers stay the same even when the editor or chat experience changes.

> Editor-based clients usually open the project folder directly. Desktop or chat-first clients may use an external MCP configuration file and then work with the project you point them to.

Reviewer: ```suggestion
Editor-based clients open the project folder directly and detect the MCP configuration from the workspace. Desktop and chat-first clients use an external configuration file and work with the project context provided in the session.
```

This overview keeps the client explanation short. The step-by-step topics below show the exact configuration locations and the exact JSON structure to use.

> ## Available Starting Paths

Reviewer: ```suggestion
## Available Setup Paths
```

> The current section includes three setup paths:

Reviewer: ```suggestion
Three setup paths are available for Ignite UI CLI MCP with {Platform}:
```

- start from a completely empty folder and let MCP help create the {Platform} project
- create the {Platform} project with Ignite UI CLI first, then continue with MCP

> - connect MCP to a {Platform} project that already exists and continue working in the current codebase

Reviewer: ```suggestion
- **Start from an empty folder** — begin with an empty workspace, add the MCP configuration, and let the AI assistant create the {Platform} project from the beginning. Use this when there is no existing project.
- **Create the project with Ignite UI CLI, then continue with MCP** — create the {Platform} project using Ignite UI CLI first, then connect MCP to the generated project. Use this when you prefer CLI scaffolding before bringing in the AI client.
- **Connect MCP to an existing project** — add the MCP configuration to a {Platform} project that already exists and continue working in the current codebase. Use this when you have an existing app and want to add Ignite UI CLI MCP and Ignite UI Theming MCP. This path also covers asking documentation and API questions from chat without making project changes.
```

The existing-project path can also be used when you want to ask documentation and API questions while working in the current app.

Each topic repeats the essential setup details so it can be used on its own, even if you open it directly from search or navigation.

> ## Open the Right Topic

Reviewer: ```suggestion
## Choose the Right Setup Path
```

Use the short guide below to choose the best starting point:

> **Start from an Empty Folder with MCP for {Platform}**: use this when you want MCP to help create the {Platform} project from the beginning.

> **Create an {Platform} Project with Ignite UI CLI, then Continue with MCP**: use this when you want to create the {Platform} project first and use MCP after that.

> **Use MCP in an Existing {Platform} Project**: use this when you already have an {Platform} project and want to connect MCP, continue working in the current codebase, and ask documentation or API questions.

- [Start from an Empty Folder with MCP for {Platform}](general-how-to-start-mcp.md)
  Use this when you want to begin with an empty workspace, add the MCP configuration for your AI client, and let the assistant help create the {Platform} project from the beginning.
- [Create a {Platform} Project with Ignite UI CLI, then Continue with MCP](general-how-to-start-mcp-cli-project.md)
  Use this when you want to create the {Platform} project with Ignite UI CLI first and continue with the generated MCP setup after that.
- [Use MCP in an Existing {Platform} Project](general-how-to-use-mcp-existing-project.md)

>   Use this when you want to connect MCP to an existing {Platform} project, continue working in the current codebase, and ask documentation or API questions in chat.

Reviewer: ```suggestion
Use the descriptions below to open the correct topic for your situation.

**[Start from an Empty Folder with Ignite UI CLI MCP for {Platform}](general-how-to-start-mcp.md)**
Use this when you want to begin with an empty workspace, add the MCP configuration, and let the assistant create the {Platform} project through prompts.

**[Create a {Platform} Project with Ignite UI CLI, then Continue with MCP](general-how-to-start-mcp-cli-project.md)**
Use this when you want to scaffold the {Platform} project with Ignite UI CLI first and connect MCP after that.

**[Use Ignite UI CLI MCP in an Existing {Platform} Project](general-how-to-use-mcp-existing-project.md)**
Use this when you have an existing {Platform} project and want to connect MCP, continue working in the current codebase, and ask documentation or API questions in chat.

## Related Topics

For more information about AI skills and theming workflows, see [{Platform} Agent Skills](ai/skills.md) and [Ignite UI Theming MCP](ai/theming-mcp.md).
```

> _"Add a new grid page to this project, wire it to sample data, and keep the current navigation structure."_

> _"What properties and events does the combo component expose, and which ones are most useful for a searchable dropdown?"_

> _"Update the existing project to add a side navigation layout and preserve the current pages and routes."_

## Prerequisites

Before configuring the MCP server, make sure you have:

- **Node.js** (v18 or later) installed - this provides the `npx` command used to launch the server.
- An **AI client with MCP support** - for example, VS Code with GitHub Copilot, GitHub Copilot cloud agent, Cursor, Claude Desktop, Claude Code, or a JetBrains IDE with the AI Assistant plugin.
- Internet access for `npx` package resolution the first time the server starts.
- One of the following starting points:
  - an empty folder for a new project
  - a project you created with Ignite UI CLI
  - an existing {ProductName} project you want to continue working on

If you want theming tools in the same client session and `igniteui-theming` is not already available in your project, run:

```bash
npm install igniteui-theming
```

> ## Setup

Reviewer: ```suggestion
## Configure Ignite UI CLI MCP for Your AI Client
```
For better GEO/AEO, that also is to be suggested for Theming MCP

        "igniteui-theming": {
          "command": "npx",
          "args": ["-y", "igniteui-theming", "igniteui-theming-mcp"]
        }
      }
    }
    ```

5. Click **OK** and restart the AI Assistant.

> [!NOTE]
> MCP support requires the AI Assistant plugin to be installed and enabled in your JetBrains IDE.

### Other MCP Clients

For any other MCP-compatible client, use the STDIO transport with these launch commands:

```bash
npx -y igniteui-cli mcp
npx -y igniteui-theming igniteui-theming-mcp
```

### GitHub

GitHub Copilot cloud agent supports repository-level MCP configuration. In your repository:

1. Open the main repository page.
2. Click **Settings**.
3. In the sidebar, go to **Copilot** then **Cloud agent**.
4. Paste your JSON in the **MCP configuration** section.
5. Click **Save**.

Use a repository configuration like this:

```json
{
  "mcpServers": {
    "igniteui-cli": {
      "type": "local",
      "command": "npx",
      "args": ["-y", "igniteui-cli", "mcp"],
      "tools": ["*"]
    },
    "igniteui-theming": {
      "type": "local",
      "command": "npx",
      "args": ["-y", "igniteui-theming", "igniteui-theming-mcp"],
      "tools": ["*"]
    }
  }

> }
> ```
> 
> This setup is useful when you want the GitHub cloud agent to use the same CLI MCP and Theming MCP tools directly from the repository context.
> 
> ## Available Tools
> 
> The MCP servers expose tools that the AI uses automatically based on your prompts. You never need to call them directly - just describe what you want.
> 
> To see the current tool list at any time, ask your AI assistant:
> 
> > _"What tools does the Ignite UI CLI MCP provide?"_
> 
> At a high level, the CLI MCP tools help with:
> 
> - creating new projects
> - working in existing projects
> - adding and modifying components
> - updating project structure and configuration
> - answering documentation and API questions

Reviewer: ```suggestion
> _"What tools does the Ignite UI CLI MCP provide?"_

At a high level, the CLI MCP tools help with:

The MCP server exposes the following tools to AI assistants:

| Tool | Description |
|------|-------------|
| `list_components` | List available Ignite UI component docs for a framework |
| `get_doc` | Fetch full markdown content of a specific component doc |
| `search_docs` | Full-text search across Ignite UI documentation |
| `get_api_reference` | Retrieve the full API reference for a component or class |
| `search_api` | Search API entries by keyword or partial component name |
| `generate_ignite_app` | Return a setup guide for a new Ignite UI project |
```
Here I expect list of the tools available form cli/readme.md

Author: Available tools have been added.

The theming server adds styling, theme, palette, and token workflows to the same client session.

<!-- React -->
For the main hands-on workflow, see [Build an App End-to-End with Ignite UI CLI MCP and Ignite UI Theming MCP](../general-how-to-mcp-e2e.md).
<!-- end: React -->
<!-- WebComponents -->
For the main hands-on workflow, see [Build an App End-to-End with Ignite UI CLI MCP and Ignite UI Theming MCP](../general-how-to-mcp-e2e.md).
<!-- end: WebComponents -->

> ## Example Scenarios

Reviewer: ```suggestion
## Common Workflows
```
For better GEO/AEO

﻿---

> title: "Ignite UI CLI MCP | Infragistics"

Reviewer: ```suggestion
title: "{Platform} Ignite UI CLI MCP | Infragistics"
```

_description: "Connect Ignite UI CLI MCP to your AI client to scaffold projects, modify existing apps, create and update components, and ask documentation questions for {ProductName}. Learn the setup options for VS Code, GitHub, Cursor, Claude Desktop, Claude Code, JetBrains, and other MCP clients."
_keywords: "{Platform}, {ProductName}, Infragistics, Ignite UI CLI MCP, Ignite UI Theming MCP, MCP, Model Context Protocol, AI, agent, GitHub Copilot, Cursor, Claude, JetBrains"
_language: en
_license: MIT
_canonicalLink: "{environment:dvUrl}/components/ai/cli-mcp"
namespace: Infragistics.Controls
mentionedTypes: []
---

# Ignite UI CLI MCP

<p class="highlight">The Ignite UI CLI MCP is a <a href="https://modelcontextprotocol.io/" target="_blank">Model Context Protocol</a> (MCP) server that enables AI assistants to scaffold projects, modify existing apps, create and update components, and answer documentation questions for {ProductName} applications. Connect the Ignite UI CLI MCP to your editor, GitHub repository, or desktop AI client and describe what you want - the assistant uses the CLI tools for you.</p>

<div class="divider"></div>

## Overview

> Instead of manually switching between CLI commands, documentation pages, and project files, you can describe the task in plain language and let the AI assistant use Ignite UI CLI MCP tools on your behalf. The server gives the AI access to project scaffolding, component generation, project modification, and documentation-aware workflows through chat or agent mode.

Reviewer: ```suggestion
Ignite UI CLI MCP gives AI assistants direct access to Ignite UI CLI project scaffolding, component generation, project modification, and documentation-aware workflows through chat or agent mode. The server works alongside Ignite UI Theming MCP — CLI MCP handles project and component workflows while Theming MCP handles palettes, themes, tokens, and styling. Most teams connect both servers in the same AI client session.
```

| `list_components`         | Lists available Ignite UI component docs for a framework. Accepts an optional keyword filter (case-insensitive substring match against name, keywords, and summary). |                                                                 |
| `get_doc`                 | Gets the full markdown content of a specific component doc by kebab-case name (e.g., `grid-editing`, `combo-overview`). Includes code samples, tables, and links.    |
| `search_docs`             | Full-text search across Ignite UI docs for a framework. Returns up to 20 ranked results with excerpts.                                                               |
| `get_project_setup_guide` | Returns the project setup guide for creating a new project in a specific framework, including CLI steps and install instructions.                                    |
| `search_api`              | Searches API entries by keyword or component name across Angular, React, and Web Components.                                                                         |

> | `get_api_reference`       | Returns the full API reference for a specific component or class, including properties, methods, and events. Supports Angular, React, and Web Components.            |

Reviewer: ```suggestion
| Tool | Description |
|------|-------------|
| `list_components` | List available Ignite UI component docs for a framework |
| `get_doc` | Fetch full markdown content of a specific component doc |
| `search_docs` | Full-text search across Ignite UI documentation |
| `get_api_reference` | Retrieve the full API reference for a component or class |
| `search_api` | Search API entries by keyword or partial component name |
| `generate_ignite_app` | Return a setup guide for a new Ignite UI project |
```

Reviewer: The list_components row had an extra empty cell — there are four pipe characters instead of three. It was a copy-paste artifact. The trailing |                                                                 | needs to be removed from that row. Every other row in the table has two cells. This will also cause MD060 misalignment since the first data row has a different structure
